### DISCUSSION TITLE
Field state does not change after trigger on controlled input.

### DISCUSSION THREAD

**Discussion by handjunyoung** [react-hook-form/react-hook-form#6353] [Q&A] [answered]
### Version Number

7.13.0

### Codesandbox/Expo snack

https://codesandbox.io/s/keen-phoebe-187b1?file=/src/App.jsx

### Steps to reproduce

1. Select the number 10 from max.
2. Enter a number between 10 and 100 in targetValue.
3. Select the number 100 from max.
4. It seems that the error message should disappear, but it remains. Is there something I am missing out on?

### Expected behaviour

The error message should disappear.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
your codesandbox has quite a lot of side effects, can you focus on the issue which is around `trigger`?

**Comment by handjunyoung** [comment, marked as answer]
Sorry.

```
<InputNumber
            style={{ width: "100%" }}
            {...inputProps}
            {...field}
            // onChange={(...args) => {
            //   field.onChange(...args);
            //   if (inputProps?.onChange) {
            //     inputProps.onChange(...args);
            //   }
            // }}
/>
```

This issue seems to occur when i try to use the onChange function for Controlled Input. Commenting this out and using a trigger when the value changes using useEffect works as intended!

Then, is it not possible to use the function at the time of change in the control input?

**Reply by bluebill1049** [reply]
this is due to `onChange` overwrite.

if you are using `controller`

```tsx
onChange={() => {
  field.onChange()
  customonChange();
}}
```

**Reply by handjunyoung** [reply]
Thank you!
